datafusion/core/tests/physical_optimizer/partition_statistics.rs

```diff
@@ -0,0 +1,317 @@
+            scan,
+        );
+        let mut sort_exec = Arc::new(sort.clone());
+        let statistics = sort_exec.statistics_by_partition()?;
+        assert_eq!(statistics.len(), 1);
+        assert_eq!(statistics[0].num_rows, Precision::Exact(4));
+        assert_eq!(statistics[0].column_statistics.len(), 2);
+        assert_eq!(statistics[0].total_byte_size, Precision::Exact(220));
+        assert_eq!(
+            statistics[0].column_statistics[0].null_count,
+            Precision::Exact(0)
+        );
+        assert_eq!(
+            statistics[0].column_statistics[0].max_value,
+            Precision::Exact(ScalarValue::Int32(Some(4)))
+        );
+        assert_eq!(
+            statistics[0].column_statistics[0].min_value,
+            Precision::Exact(ScalarValue::Int32(Some(1)))
+        );
+        sort_exec = Arc::new(sort.with_preserve_partitioning(true));
+        let statistics = sort_exec.statistics_by_partition()?;
+        assert_eq!(statistics.len(), 2);
+        assert_eq!(statistics[0].num_rows, Precision::Exact(2));
+        assert_eq!(statistics[1].num_rows, Precision::Exact(2));
+        assert_eq!(statistics[0].column_statistics.len(), 2);
+        assert_eq!(statistics[1].column_statistics.len(), 2);
+        assert_eq!(statistics[0].total_byte_size, Precision::Exact(110));
+        assert_eq!(statistics[1].total_byte_size, Precision::Exact(110));
+        assert_eq!(
+            statistics[0].column_statistics[0].null_count,
+            Precision::Exact(0)
+        );
+        assert_eq!(
+            statistics[0].column_statistics[0].max_value,
+            Precision::Exact(ScalarValue::Int32(Some(4)))
+        );
+        assert_eq!(
+            statistics[0].column_statistics[0].min_value,
+            Precision::Exact(ScalarValue::Int32(Some(3)))
+        );
+        assert_eq!(
+            statistics[1].column_statistics[0].null_count,
+            Precision::Exact(0)
+        );
+        assert_eq!(
+            statistics[1].column_statistics[0].max_value,
+            Precision::Exact(ScalarValue::Int32(Some(2)))
+        );
+        assert_eq!(
+            statistics[1].column_statistics[0].min_value,
+            Precision::Exact(ScalarValue::Int32(Some(1)))
+        );
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_statistics_by_partition_of_filter() -> datafusion_common::Result<()> {
+        let scan = generate_listing_table_with_statistics(Some(2)).await;
+        let schema = Schema::new(vec![Field::new("id", DataType::Int32, false)]);
+        let predicate = binary(
+            Arc::new(Column::new("id", 0)),
+            Operator::Lt,
+            lit(1i32),
+            &schema,
+        )?;
+        let filter: Arc<dyn ExecutionPlan> =
+            Arc::new(FilterExec::try_new(predicate, scan)?);
+        let _full_statistics = filter.statistics()?;
+        // The full statistics is invalid, at least, we can improve the selectivity estimation of the filter
```

Comment:
I don't understand this comment. Should we file a ticket to track whatever the expected result is?

Reply:
I think the `full_statistics` can be improved, I'll open an issue to follow the further improvement.